website/markdown/apps-at-scale/2020-06-22-kamil/post.mdx

```diff
@@ -0,0 +1,87 @@
+---
+layout: post
+title: 'With Bazel we were able to reduce build times by 95% on clean builds - Kamil Pyć'
+categories:
+  [
+    Tuist,
+    kamil,
+    soundcloud,
+    app development at scale,
+    xcodeproj,
+    xcode,
+    swift,
+    Allegro,
+  ]
+excerpt: "In this interview, we talk with Kamil Pyć, Senior Mobile Developer at Allegro. Allegro is one of the few companies that have undertaken replacing Xcode's build system with Bazel, and that led them to an improvement in build times of roughly 95%. In this interview, Kamil shares more about Bazel's adoption, and some other insights about their project and teams."
+header_image: https://unsplash.com/photos/OnafBB6V_PM/download?force=true&w=2400
+interviewee_name: 'Kamil Pyć'
+interviewee_twitter_handle: 'KamilPyc'
+interviewee_avatar: https://pbs.twimg.com/profile_images/745018488184516608/Skzxzejl_400x400.jpg
+community_topic: https://community.tuist.io/t/kamil-pycs-interview/74
+---
+
+When we announced the idea of publishing a series of interviews around app development at scale Kamil offered to participate and tell us more about iOS development in the biggest e-commerce company in Poland, [Allegro](https://allegro.pl)
+
+### Tell us more about yourself and the project you work on
+
+I'm Kamil Pyć. I have been working as a Senior Mobile Developer at [Allegro](https://allegro.pl) for over 6 years. Currently I am part of the team responsible for supporting other developers - providing tooling and solutions for them to create our app fast and efficiently. Allegro is Poland's biggest e-commerce company - providing millions of products to millions of customers. That was a succulent offer that we couldn't turn down so let's dive right into the interview with Kamil:
+
+## Team structure
+
+### How many engineers work on features and how many take care of the infrastructure of your projects?
+
+We used to have platform specific teams but now mobile developers are part of feature teams. Currently we have about **35 developers** working on our iOS application. Our app is divided into **50 modules** so every team has at least one module to take care of. We have a dedicated team that performs releases every two weeks and our iOS infrastructure is a team of three people.
+
+## Project and code architecture
+
+### What are the main challenges on your architecture when scaling?
+
+There is nothing revolutionary about our code architecture - we mostly use MVVM, but if there is part of app where other paradigms would fit better we can use them. For example reactive programming architecture is applied just in the module of user registration which has a lot of input and validation.
```

Comment:
This line is a bit unclear to me: _but if there is part of app where other paradigms would fit better we can use them_

Not sure if the phrasing is correct.

Reply:
I re-phrased it a little bit